### ISSUE TITLE
Report kernel execution information.

### ISSUE THREAD

**Issue by rhornung67** [LLNL/RAJA#743]
Explore possible mechanisms to return useful information about how a kernel was launched, error codes, etc. from forall and kernel methods.

At a most basic level, it would be helpful to RAJA users if we returned CUDA error codes, etc. after a kernel executes so they can optionally inspect this information. In addition, we could return a data structure that includes info about how a kernel was launched to see if actual execution matched users' expectations. We could probably find useful ways to provide useful information for CPU kernel launch as well.

**Comment by davidbeckingsale** [comment]
We might be able to wrap this up with the `camp::resource` stuff? @trws @mdavis36

**Comment by rhornung67** [comment]
If we could, that would be great as it could be used in other projects more easily.

**Comment by mdavis36** [comment]
@davidbeckingsale Agreed, especially since the returned event types are already platform specific. With CUDA we could call cudaPeakAtLastError after the kernel then pass back a cudaError_t within the event that the user could then query.

**Comment by trws** [comment]
I had at least a comment, if not an issue, for wanting to add a mechanism to do something like this in the `resource` stuff already actually, it needs it to be able to do something sensible when transfers or event waits receive the failure, so I think that makes perfect sense at least for error codes.  We might want to have some extra machinery in the raja end to capture the kernel launch information that Rich is talking about though, since that's separate, or have a way to insert some ordered state information into it so you can track back to what produced the error.
